ui/page.go

```diff
@@ -473,37 +472,27 @@ func (page pageCommon) Layout(gtx layout.Context, body layout.Widget) layout.Dim
 		}),
 		layout.Stacked(func(gtx C) D {
 			// global toasts. Stack toast on all pages and contents
-			t := func(n *toast) layout.Dimensions {
-				return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
-					layout.Flexed(1, func(gtx C) D {
-						return layout.Center.Layout(gtx, func(gtx C) D {
-							return layout.Inset{Top: values.MarginPadding65}.Layout(gtx, func(gtx C) D {
-								return displayToast(page.theme, gtx, n)
+			if len(*page.toasts) == 0 {
+				return layout.Dimensions{}
+			}
+			gtx.Constraints.Min.X = gtx.Constraints.Max.X
+			return layout.Center.Layout(gtx, func(gtx C) D {
+				return page.toastList.Layout(gtx, len(*page.toasts), func(gtx C, index int) D {
+					t := func(n *toast) layout.Dimensions {
+						inset := layout.Inset{Top: values.MarginPadding20}
+						if index == 0 {
+							inset.Top = values.MarginPadding65
+						}
+						return inset.Layout(gtx, func(gtx C) D {
+							(*page.toasts)[index].Timer(time.Second*3, func() {
+								*page.toasts = (*page.toasts)[1:]
 							})
+							return displayToast(page.theme, gtx, n)
 						})
-					}),
-				)
-			}
-
-		outer:
-			for {
-				select {
-				case n := <-page.toast:
-					page.toastLoad.success = n.success
-					page.toastLoad.text = n.text
-					page.toastLoad.ResetTimer()
-				default:
-					break outer
-				}
-			}
-
-			if page.toastLoad.text != "" {
-				page.toastLoad.Timer(time.Second*3, func() {
-					page.toastLoad.text = ""
+					}
+					return t((*page.toasts)[index])
```

Comment:
You can simply loop through the list of toasts, call the time.AfterFunc method to remove that toast from the list in the specified duration. You should add a mutex lock for appending and removing data from the toast list so a data race issue doesn't occur when trying to access data in the list.